near-sdk/src/collections/legacy_tree_map.rs

```diff
@@ -585,6 +585,16 @@ where
         self.save(&n);
         self.tree.pop();
     }
+
+    pub fn to_tree_map<S: IntoStorageKey>(self, prefix: S) -> super::TreeMap<K, V> {
+        // We read this into the heap first so one can use the same prefix without getting into trouble
+        let elements: Vec<(K, V)> = self.into_iter().collect();
```

Comment:
So I'm curious about if this should be clearing the state when doing this conversion.

Seems like if the API is `(self) -> Self` it would be intuitive that the storage would be cleared and the new collection would be filled with this data.

Otherwise, we could consider this pattern of not clearing and have the method be something like:

```rust
fn copy_to_tree_map(&self, ..) -> super::TreeMap...
```

with a better name and whatnot.

Also, the other part that might seem opinionated is if the prefix is added as a parameter here. This makes sense if we go the copy route, but if the API is for an owned `self`, it might be more intuitive to just remove the prefix and re-use the one from which you're clearing.

To summarize, my intuitions are that this should either:
- Remove the prefix from parameters, re-use and clear the state, put state back in at the same prefix
- Change this API to `&self` and make it clear that the elements are not being removed from storage
  - Clearly document this fact and make it clear to devs that they should not use the same prefix as before (seems like a footgun and also requires devs to manually `clear` the state of the old collection)
- Another option for fun: Just create a `drain` iterator that clears and gives all key-value pairs back. Let the user decide if they want to clear the elements and put it into another collection or to just use the current iterator and use that to just fill another collection
  - This gives more freedom to migrate this state into other data structures, rather than just the `collections::ltm` -> `collections::tm` path
  
Seems like it might be good to get another opinion on this. @itegulov, you might be most familiar with these APIs, wdyt?